### PR TITLE
Support ruff, remove flake8, and fix nox versioning, config in pyproject.toml

### PR DESCRIPTION
**Pull request by Lunarmagpie**
This PR cleans up CI by doing the stuff in the title

### CODE REVIEW

**Comment by MithicSpirit** [comment]
We should add more python versions to the CI imo.

File: mypy.ini [hunk 1]
```diff
@@ -1,11 +0,0 @@
-[mypy]
-warn_return_any=True
-warn_unused_configs=True
-strict=True
-namespace_packages=True
-
-[mypy-nox.*]
-ignore_missing_imports=True
```
Comment (MithicSpirit) [inline mypy.ini:8]: What happened to `ignore-missing-imports` for the nox mypy module?

File: .isort.cfg [hunk 2]
```diff
@@ -1,3 +0,0 @@
-[settings]
-profile = black
-line_length = 79
```
Comment (MithicSpirit) [inline .isort.cfg:3]: What happened to the line length limit in the isort configuration?

**Comment by CircuitSacul** [comment]
> We should add more python versions to the CI imo, just to make sure that every version works correctly.

linting is going to be the same regardless of python version. For type-checking, aren't 3.* python versions backwards compatible?

File: .isort.cfg (hunk 2 above)
Comment (Lunarmagpie) [reply .isort.cfg:3]: Pretty sure isn't needed because `profile=black` sets it to the right one

File: mypy.ini (hunk 1 above)
Comment (Lunarmagpie) [reply mypy.ini:8]: It wasn't doing anything. Nox is typed so im not sure why that was there to begin with.

File: .isort.cfg (hunk 2 above)
Comment (CircuitSacul) [reply .isort.cfg:3]: > Pretty sure isn't needed because `profile=black` sets it to the right one

Black uses 88 by default

File: mypy.ini (hunk 1 above)
Comment (CircuitSacul) [reply mypy.ini:8]: CI passes so it clearly has types

File: .isort.cfg (hunk 2 above)
Comment (Lunarmagpie) [reply .isort.cfg:3]: It copies the line length for black. I just checked and isort complains about a length 83 line with the config i set up.

Comment (CircuitSacul) [reply .isort.cfg:3]: Oh nice, so it actually looks for the black config.

**Comment by Lunarmagpie** [comment]
> linting is going to be the same regardless of python version. For type-checking, aren't 3.* python versions backwards compatible?

Agree with this. I think we a script to import `mCodingBot` or tests to check that it actually works on multiple python version.

**Review by MithicSpirit — APPROVED** [review]
lgtm.

**Review by trag1c — APPROVED** [review]

**Pull request merged**